handler.lua

```diff
@@ -9,11 +9,20 @@ local is_json_body = header_filter.is_json_body
 
 local cjson_decode = require("cjson").decode
 local cjson_encode = require("cjson").encode
+local CACHE_HEADER = 'X-Kong-Cache-Status'
 
-local function cacheable_request(method, uri, conf)
+local status_code_bypass = { ngx.HTTP_BAD_GATEWAY, ngx.HTTP_SERVICE_UNAVAILABLE, ngx.HTTP_GATEWAY_TIMEOUT, ngx.HTTP_INTERNAL_SERVER_ERROR, ngx.HTTP_METHOD_NOT_IMPLEMENTED }
+
+local function cacheable_request(method, uri, conf, status_code)
   if method ~= "GET" then
     return false
   end
+  
+  for _,v in ipairs(status_code_bypass) do
+    if v == status_code then
```

Comment:
Instead of whitelisting status_codes, maybe we should do something like
```
if status_code >= 300 then
  return false
end
```

Reply:
Makes sense. I was figuring if we can make this a part of config as well. That'll leave the status_code config upto the user. Thoughts?